### ISSUE TITLE
3.0.0-snapshot-441 - both("HasFriend").size()  not working

### ISSUE THREAD

**Issue by santo-it** [orientechnologies/orientdb#7249]
### OS: win 10

--

demodb v. 0.75

SELECT @rid, out("HasFriend").size() AS OutFriendsNumber, in("HasFriend").size() AS InFriendsNumber, both("HasFriend").size() as FriendTotalMethod1, (out("HasFriend").size() + in("HasFriend").size()) as FriendTotalMethod2 FROM `Profiles` LIMIT 10

![luigi_github](https://cloud.githubusercontent.com/assets/14923389/24085773/101d795c-0d02-11e7-86a3-017583787742.PNG)

works fine in 2.2.

please add test

Thanks,

**Comment by luigidellaquila** [comment]
Fixed in develop branch

Thanks

Luigi

**Comment by santo-it** [comment]
Hi,

Thanks for fixing the rid problem. This specific issue however is about both("EdgeClass").size() not working. I just noticed subject of the issue was wrong, I fixed it now.  The issue is : for some reason both("EdgeClass").size() always returns 2

Check the result of this query:

SELECT both("HasFriend").size() as FriendTotalMethod1, (out("HasFriend").size() + in("HasFriend").size()) as FriendTotalMethod2 FROM Profiles LIMIT 10

FriendTotalMethod1 <> FriendTotalMethod2 but they should be the same

![luigi_github2](https://cloud.githubusercontent.com/assets/14923389/24171488/fae0adfa-0e84-11e7-8694-4e40d530c176.PNG)

Reopening,

Thanks,
